Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/MinimumFlowBypass/Controller.mo

```diff
@@ -0,0 +1,223 @@
+﻿within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.MinimumFlowBypass;
+block Controller
+  "Controller for chilled water minimum flow bypass valve"
+
+  parameter Integer nChi=3 "Total number of chillers";
+  parameter Modelica.SIunits.Time byPasSetTim = 300
+    "Time to reset minimum by-pass flow";
+  parameter Modelica.SIunits.VolumeFlowRate minFloSet[nChi] = {0.005, 0.005, 0.005}
+    "Minimum chilled water flow through each chiller";
+  parameter Modelica.SIunits.VolumeFlowRate maxFloSet[nChi]={0.025,0.025,0.025}
+    "Maximum chilled water flow through each chiller";
+  parameter Buildings.Controls.OBC.CDL.Types.SimpleController controllerType=
+    Buildings.Controls.OBC.CDL.Types.SimpleController.PID
+    "Type of controller"
+    annotation (Dialog(group="Controller"));
+  parameter Real k=1 "Gain of controller"
+    annotation (Dialog(group="Controller"));
+  parameter Modelica.SIunits.Time Ti=0.5 "Time constant of integrator block"
+    annotation (Dialog(group="Controller"));
+  parameter Modelica.SIunits.Time Td=0.1 "Time constant of derivative block"
+    annotation (Dialog(group="Controller"));
+  parameter Real yMax=1 "Upper limit of output"
+    annotation (Dialog(group="Controller"));
+  parameter Real yMin=0.1 "Lower limit of output"
+    annotation (Dialog(group="Controller"));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uChiWatPum
+    "Maximum status feedback of all the chilled water pumps: true means at least one pump is proven on"
+    annotation (Placement(transformation(extent={{-140,110},{-100,150}}),
+      iconTransformation(extent={{-120,80},{-100,100}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput VChiWat_flow(
+    final min=0,
+    final unit="m3/s",
+    quantity="VolumeFlowRate")
+    "Measured chilled water flow rate through chillers"
+    annotation (Placement(transformation(extent={{-140,70},{-100,110}}),
+      iconTransformation(extent={{-120,60},{-100,80}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uStaUp "Stage up logical signal"
+    annotation (Placement(transformation(extent={{-140,40},{-100,80}}),
+      iconTransformation(extent={{-120,40},{-100,60}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uUpsDevSta
+    "Status of resetting status of device before reset minimum bypass flow setpoint"
+    annotation (Placement(transformation(extent={{-140,10},{-100,50}}),
+      iconTransformation(extent={{-120,20},{-100,40}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uEnaNexChi
+    "Status to indicate that it starts to enable another chiller. This input used when the stage change needs chiller on/off"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),
+      iconTransformation(extent={{-120,0},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uOnOff
+    "Indicate if the stage require one chiller to be enabled while another is disabled"
```

Comment:
@AntoineGautier there will be another external sequence to supply the input. 
The signal is more related to the staging process. It will be applied to both stage up and down process and the chiller status will not change before the minimum flow has been reset. So we cannot simply use chillers on/off command.